### ISSUE TITLE
This apiVersion and/or kind does not reference a schema known by Cloud Code.

### ISSUE THREAD

**Issue by vinnytwice** [mongodb/mongodb-kubernetes-operator#1060]
**What did you do to encounter the bug?**
I'm using docker desktop on an MacBook Air M1 and I just installed the MongoDBCommunity operator following the guide here https://github.com/mongodb/mongodb-kubernetes-operator/blob/master/docs/install-upgrade.md using the kubectl method.
Then I went to VS Code project and pasted a ReplicaSet following the guide here https://github.com/mongodb/mongodb-kubernetes-operator/blob/master/docs/deploy-configure.md in a .yaml file but I get an error: 
`This apiVersion and/or kind does not reference a schema known by Cloud Code. Please ensure you are using a valid apiVersion and kind.`

if run the command `kubectl api-resources` I see it is installed
```
mongodbcommunity                  mdbc         mongodbcommunity.mongodb.com/v1        true         MongoDBCommunity
```
As a test I commented out everything and re added one line at the time.`apiVersion: mongodbcommunity.mongodb.com/v1` is accepted, but as soon as `kind: MongoDBCommunity` is added they both get underlined.

Install procedure used.. 

- Cloned the repo, and cd'ed into it.
(Configure the Operator to watch other namespaces.)
- edited the manager file  /config/manager/manager.yaml
```
env:
        - name: WATCH_NAMESPACE
          value: "*"
          # valueFrom:
          #   fieldRef:
          #     fieldPath: metadata.namespace
```
- edited the file in /deploy/clusterwide/cluster_role_binding.yaml
```
- kind: ServiceAccount
  namespace: default
  name: mongodb-kubernetes-operator
```
- applied it 
```
kubectl apply -f deploy/clusterwide
```
- deploy a Role, RoleBinding and ServiceAccount in every namespace
```
kubectl apply -k config/rbac --namespace default
```

- Left untouched image and url for mongo in /config/manager/manager.yaml
```
- name: MONGODB_IMAGE
          value: mongo
        - name: MONGODB_REPO_URL
          value: docker.io
```

(Install)

- edited config/crd/bases/mongodbcommunity.mongodb.com_mongodbcommunity.yaml
```
metadata:
  annotations:
    controller-gen.kubebuilder.io/version: v0.7.4  (this was a different version)
  creationTimestamp: null
  name: mongodbcommunity.mongodbcommunity.mongodb.com
```

- applied it
```
kubectl apply -f config/crd/bases/mongodbcommunity.mongodb.com_mongodbcommunity.yaml
```
- Installed the necessary roles and role-bindings
```
kubectl apply -k config/rbac/ --namespace default
```

- Install the Operator
```
kubectl create -f config/manager/manager.yaml --namespace default
```

I went through the install process again and at each step verified the the step was applied correctly
```
vincenzocalia@vincenzos-MacBook-Air server-node % kubectl get crd/mongodbcommunity.mongodbcommunity.mongodb.com
NAME                                            CREATED AT
mongodbcommunity.mongodbcommunity.mongodb.com   2022-07-28T13:09:48Z
vincenzocalia@vincenzos-MacBook-Air server-node % kubectl get role mongodb-kubernetes-operator            
NAME                          CREATED AT
mongodb-kubernetes-operator   2022-07-28T13:10:28Z
vincenzocalia@vincenzos-MacBook-Air server-node % kubectl get role mongodb-kubernetes-operator --namespace default

NAME                          CREATED AT
mongodb-kubernetes-operator   2022-07-28T13:10:28Z
vincenzocalia@vincenzos-MacBook-Air server-node % kubectl get rolebinding mongodb-kubernetes-operator             
NAME                          ROLE                               AGE
mongodb-kubernetes-operator   Role/mongodb-kubernetes-operator   4h34m
vincenzocalia@vincenzos-MacBook-Air server-node % kubectl get rolebinding mongodb-kubernetes-operator --namespace default
NAME                          ROLE                               AGE
mongodb-kubernetes-operator   Role/mongodb-kubernetes-operator   4h34m
vincenzocalia@vincenzos-MacBook-Air server-node % kubectl get serviceaccount mongodb-kubernetes-operator         
NAME                          SECRETS   AGE
mongodb-kubernetes-operator   0         4h34m
vincenzocalia@vincenzos-MacBook-Air server-node % kubectl get serviceaccount mongodb-kubernetes-operator --namespace default
NAME                          SECRETS   AGE
mongodb-kubernetes-operator   0         4h34m
vincenzocalia@vincenzos-MacBook-Air server-node % kubectl get pods            
NAME                                           READY   STATUS    RESTARTS   AGE
mongodb-kubernetes-operator-7646658db4-8cnvn   1/1     Running   0          63m
vincenzocalia@vincenzos-MacBook-Air server-node % kubectl get pods --namespace default
NAME                                           READY   STATUS    RESTARTS   AGE
mongodb-kubernetes-operator-7646658db4-8cnvn   1/1     Running   0          63m
vincenzocalia@vincenzos-MacBook-Air server-node % kubectl get sts
No resources found in default namespace.
vincenzocalia@vincenzos-MacBook-Air server-node % kubectl api-resources
NAME                              SHORTNAMES   APIVERSION                             NAMESPACED   KIND
bindings                                       v1                                     true         Binding
componentstatuses                 cs           v1                                     false        ComponentStatus
configmaps                        cm           v1                                     true         ConfigMap
endpoints                         ep           v1                                     true         Endpoints
events                            ev           v1                                     true         Event
limitranges                       limits       v1                                     true         LimitRange
namespaces                        ns           v1                                     false        Namespace
nodes                             no           v1                                     false        Node
persistentvolumeclaims            pvc          v1                                     true         PersistentVolumeClaim
persistentvolumes                 pv           v1                                     false        PersistentVolume
pods                              po           v1                                     true         Pod
podtemplates                                   v1                                     true         PodTemplate
replicationcontrollers            rc           v1                                     true         ReplicationController
resourcequotas                    quota        v1                                     true         ResourceQuota
secrets                                        v1                                     true         Secret
serviceaccounts                   sa           v1                                     true         ServiceAccount
services                          svc          v1                                     true         Service
mutatingwebhookconfigurations                  admissionregistration.k8s.io/v1        false        MutatingWebhookConfiguration
validatingwebhookconfigurations                admissionregistration.k8s.io/v1        false        ValidatingWebhookConfiguration
customresourcedefinitions         crd,crds     apiextensions.k8s.io/v1                false        CustomResourceDefinition
apiservices                                    apiregistration.k8s.io/v1              false        APIService
controllerrevisions                            apps/v1                                true         ControllerRevision
daemonsets                        ds           apps/v1                                true         DaemonSet
deployments                       deploy       apps/v1                                true         Deployment
replicasets                       rs           apps/v1                                true         ReplicaSet
statefulsets                      sts          apps/v1                                true         StatefulSet
tokenreviews                                   authentication.k8s.io/v1               false        TokenReview
localsubjectaccessreviews                      authorization.k8s.io/v1                true         LocalSubjectAccessReview
selfsubjectaccessreviews                       authorization.k8s.io/v1                false        SelfSubjectAccessReview
selfsubjectrulesreviews                        authorization.k8s.io/v1                false        SelfSubjectRulesReview
subjectaccessreviews                           authorization.k8s.io/v1                false        SubjectAccessReview
horizontalpodautoscalers          hpa          autoscaling/v2                         true         HorizontalPodAutoscaler
cronjobs                          cj           batch/v1                               true         CronJob
jobs                                           batch/v1                               true         Job
certificatesigningrequests        csr          certificates.k8s.io/v1                 false        CertificateSigningRequest
leases                                         coordination.k8s.io/v1                 true         Lease
endpointslices                                 discovery.k8s.io/v1                    true         EndpointSlice
events                            ev           events.k8s.io/v1                       true         Event
flowschemas                                    flowcontrol.apiserver.k8s.io/v1beta2   false        FlowSchema
prioritylevelconfigurations                    flowcontrol.apiserver.k8s.io/v1beta2   false        PriorityLevelConfiguration
mongodbcommunity                  mdbc         mongodbcommunity.mongodb.com/v1        true         MongoDBCommunity
ingressclasses                                 networking.k8s.io/v1                   false        IngressClass
ingresses                         ing          networking.k8s.io/v1                   true         Ingress
networkpolicies                   netpol       networking.k8s.io/v1                   true         NetworkPolicy
runtimeclasses                                 node.k8s.io/v1                         false        RuntimeClass
poddisruptionbudgets              pdb          policy/v1                              true         PodDisruptionBudget
podsecuritypolicies               psp          policy/v1beta1                         false        PodSecurityPolicy
clusterrolebindings                            rbac.authorization.k8s.io/v1           false        ClusterRoleBinding
clusterroles                                   rbac.authorization.k8s.io/v1           false        ClusterRole
rolebindings                                   rbac.authorization.k8s.io/v1           true         RoleBinding
roles                                          rbac.authorization.k8s.io/v1           true         Role
priorityclasses                   pc           scheduling.k8s.io/v1                   false        PriorityClass
csidrivers                                     storage.k8s.io/v1                      false        CSIDriver
csinodes                                       storage.k8s.io/v1                      false        CSINode
csistoragecapacities                           storage.k8s.io/v1                      true         CSIStorageCapacity
storageclasses                    sc           storage.k8s.io/v1                      false        StorageClass
volumeattachments                              storage.k8s.io/v1                      false        VolumeAttachment
```
it looks all good but still get the error..

Now there is one step 
>For each namespace that you want the Operator to watch, run the following commands to deploy a Role, RoleBinding and ServiceAccount in that namespace:
```
kubectl apply -k config/rbac --namespace <my-namespace>
```
I did apply it only to `default` namespace but as these are my spacenames
```
default           Active   12d
ingress-nginx     Active   10d
kube-node-lease   Active   12d
kube-public       Active   12d
kube-system       Active   12d
```
should apply it to all of them??

What can I check to see if I did install it properly or not? 

Thank you very much.
Cheers

**What did you expect?**
resource should be recognised

**What happened instead?**
as soon as I add a kind: property to the yaml file I get the `This apiVersion and/or kind does not reference a schema known by Cloud Code.` error.

**Screenshots**
![Screenshot 2022-07-28 at 20 27 54](https://user-images.githubusercontent.com/32189706/181611963-907844b8-c310-4b87-9149-3db1f7f56c65.png)

![Screenshot 2022-07-28 at 20 28 21](https://user-images.githubusercontent.com/32189706/181612005-ab46b6aa-5d84-4195-b943-86669031ed49.png)

**Operator Information**
 - Operator Version: 0.7.4
 - MongoDB Image used: mongo 
 - docker.io
**Kubernetes Cluster Information**
 - Client Version: v1.24.1
 - Server Version: v1.24.1
 - Kustomize Version: v4.5.4

If possible, please include:
 - `kubectl describe` output
 - yaml definitions for your objects
 - log files for the operator and database pods
# mongodbcommunity.mongodb.com_mongodbcommunity.yaml:
```
apiVersion: apiextensions.k8s.io/v1
kind: CustomResourceDefinition
metadata:
  annotations:
    controller-gen.kubebuilder.io/version: v0.7.4
  creationTimestamp: null
  name: mongodbcommunity.mongodbcommunity.mongodb.com
spec:
  group: mongodbcommunity.mongodb.com
  names:
    kind: MongoDBCommunity
    listKind: MongoDBCommunityList
    plural: mongodbcommunity
    shortNames:
    - mdbc
    singular: mongodbcommunity
  scope: Namespaced
  versions:
  - additionalPrinterColumns:
    - description: Current state of the MongoDB deployment
      jsonPath: .status.phase
      name: Phase
      type: string
    - description: Version of MongoDB server
      jsonPath: .status.version
      name: Version
      type: string
    name: v1
    schema:
      openAPIV3Schema:
        description: MongoDBCommunity is the Schema for the mongodbs API
        properties:
          apiVersion:
            description: 'APIVersion defines the versioned schema of this representation
              of an object. Servers should convert recognized schemas to the latest
              internal value, and may reject unrecognized values. More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#resources'
            type: string
          kind:
            description: 'Kind is a string value representing the REST resource this
              object represents. Servers may infer this from the endpoint the client
              submits requests to. Cannot be updated. In CamelCase. More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#types-kinds'
            type: string
          metadata:
            type: object
          spec:
            description: MongoDBCommunitySpec defines the desired state of MongoDB
            properties:
              additionalMongodConfig:
                description: 'AdditionalMongodConfig is additional configuration that
                  can be passed to each data-bearing mongod at runtime. Uses the same
                  structure as the mongod configuration file: https://www.mongodb.com/docs/manual/reference/configuration-options/'
                nullable: true
                type: object
                x-kubernetes-preserve-unknown-fields: true
              arbiters:
                description: 'Arbiters is the number of arbiters to add to the Replica
                  Set. It is not recommended to have more than one arbiter per Replica
                  Set. More info: https://www.mongodb.com/docs/manual/tutorial/add-replica-set-arbiter/'
                type: integer
              automationConfig:
                description: AutomationConfigOverride is merged on top of the operator
                  created automation config. Processes are merged by name. Currently
                  Only the process.disabled field is supported.
                properties:
                  processes:
                    items:
                      description: OverrideProcess contains fields that we can override
                        on the AutomationConfig processes.
                      properties:
                        disabled:
                          type: boolean
                        name:
                          type: string
                      required:
                      - disabled
                      - name
                      type: object
                    type: array
                required:
                - processes
                type: object
              featureCompatibilityVersion:
                description: FeatureCompatibilityVersion configures the feature compatibility
                  version that will be set for the deployment
                type: string
              members:
                description: Members is the number of members in the replica set
                type: integer
              prometheus:
                description: Prometheus configurations.
                properties:
                  metricsPath:
                    description: Indicates path to the metrics endpoint.
                    pattern: ^\/[a-z0-9]+$
                    type: string
                  passwordSecretRef:
                    description: Name of a Secret containing a HTTP Basic Auth Password.
                    properties:
                      key:
                        description: Key is the key in the secret storing this password.
                          Defaults to "password"
                        type: string
                      name:
                        description: Name is the name of the secret storing this user's
                          password
                        type: string
                    required:
                    - name
                    type: object
                  port:
                    description: Port where metrics endpoint will bind to. Defaults
                      to 9216.
                    type: integer
                  tlsSecretKeyRef:
                    description: Name of a Secret (type kubernetes.io/tls) holding
                      the certificates to use in the Prometheus endpoint.
                    properties:
                      key:
                        description: Key is the key in the secret storing this password.
                          Defaults to "password"
                        type: string
                      name:
                        description: Name is the name of the secret storing this user's
                          password
                        type: string
                    required:
                    - name
                    type: object
                  username:
                    description: HTTP Basic Auth Username for metrics endpoint.
                    type: string
                required:
                - passwordSecretRef
                - username
                type: object
              replicaSetHorizons:
                description: ReplicaSetHorizons Add this parameter and values if you
                  need your database to be accessed outside of Kubernetes. This setting
                  allows you to provide different DNS settings within the Kubernetes
                  cluster and to the Kubernetes cluster. The Kubernetes Operator uses
                  split horizon DNS for replica set members. This feature allows communication
                  both within the Kubernetes cluster and from outside Kubernetes.
                items:
                  additionalProperties:
                    type: string
                  type: object
                type: array
              security:
                description: Security configures security features, such as TLS, and
                  authentication settings for a deployment
                properties:
                  authentication:
                    properties:
                      ignoreUnknownUsers:
                        default: true
                        nullable: true
                        type: boolean
                      modes:
                        description: Modes is an array specifying which authentication
                          methods should be enabled.
                        items:
                          enum:
                          - SCRAM
                          - SCRAM-SHA-256
                          - SCRAM-SHA-1
                          type: string
                        type: array
                    required:
                    - modes
                    type: object
                  roles:
                    description: User-specified custom MongoDB roles that should be
                      configured in the deployment.
                    items:
                      description: CustomRole defines a custom MongoDB role.
                      properties:
                        authenticationRestrictions:
                          description: The authentication restrictions the server
                            enforces on the role.
                          items:
                            description: AuthenticationRestriction specifies a list
                              of IP addresses and CIDR ranges users are allowed to
                              connect to or from.
                            properties:
                              clientSource:
                                items:
                                  type: string
                                type: array
                              serverAddress:
                                items:
                                  type: string
                                type: array
                            required:
                            - clientSource
                            - serverAddress
                            type: object
                          type: array
                        db:
                          description: The database of the role.
                          type: string
                        privileges:
                          description: The privileges to grant the role.
                          items:
                            description: Privilege defines the actions a role is allowed
                              to perform on a given resource.
                            properties:
                              actions:
                                items:
                                  type: string
                                type: array
                              resource:
                                description: Resource specifies specifies the resources
                                  upon which a privilege permits actions. See https://www.mongodb.com/docs/manual/reference/resource-document
                                  for more.
                                properties:
                                  anyResource:
                                    type: boolean
                                  cluster:
                                    type: boolean
                                  collection:
                                    type: string
                                  db:
                                    type: string
                                type: object
                            required:
                            - actions
                            - resource
                            type: object
                          type: array
                        role:
                          description: The name of the role.
                          type: string
                        roles:
                          description: An array of roles from which this role inherits
                            privileges.
                          items:
                            description: Role is the database role this user should
                              have
                            properties:
                              db:
                                description: DB is the database the role can act on
                                type: string
                              name:
                                description: Name is the name of the role
                                type: string
                            required:
                            - db
                            - name
                            type: object
                          type: array
                      required:
                      - db
                      - privileges
                      - role
                      type: object
                    type: array
                  tls:
                    description: TLS configuration for both client-server and server-server
                      communication
                    properties:
                      caCertificateSecretRef:
                        description: CaCertificateSecret is a reference to a Secret
                          containing the certificate for the CA which signed the server
                          certificates The certificate is expected to be available
                          under the key "ca.crt"
                        properties:
                          name:
                            type: string
                        required:
                        - name
                        type: object
                      caConfigMapRef:
                        description: CaConfigMap is a reference to a ConfigMap containing
                          the certificate for the CA which signed the server certificates
                          The certificate is expected to be available under the key
                          "ca.crt" This field is ignored when CaCertificateSecretRef
                          is configured
                        properties:
                          name:
                            type: string
                        required:
                        - name
                        type: object
                      certificateKeySecretRef:
                        description: CertificateKeySecret is a reference to a Secret
                          containing a private key and certificate to use for TLS.
                          The key and cert are expected to be PEM encoded and available
                          at "tls.key" and "tls.crt". This is the same format used
                          for the standard "kubernetes.io/tls" Secret type, but no
                          specific type is required. Alternatively, an entry tls.pem,
                          containing the concatenation of cert and key, can be provided.
                          If all of tls.pem, tls.crt and tls.key are present, the
                          tls.pem one needs to be equal to the concatenation of tls.crt
                          and tls.key
                        properties:
                          name:
                            type: string
                        required:
                        - name
                        type: object
                      enabled:
                        type: boolean
                      optional:
                        description: Optional configures if TLS should be required
                          or optional for connections
                        type: boolean
                    required:
                    - enabled
                    type: object
                type: object
              statefulSet:
                description: StatefulSetConfiguration holds the optional custom StatefulSet
                  that should be merged into the operator created one.
                properties:
                  spec:
                    type: object
                    x-kubernetes-preserve-unknown-fields: true
                required:
                - spec
                type: object
              type:
                description: Type defines which type of MongoDB deployment the resource
                  should create
                enum:
                - ReplicaSet
                type: string
              users:
                description: Users specifies the MongoDB users that should be configured
                  in your deployment
                items:
                  properties:
                    connectionStringSecretName:
                      description: ConnectionStringSecretName is the name of the secret
                        object created by the operator which exposes the connection
                        strings for the user. If provided, this secret must be different
                        for each user in a deployment.
                      type: string
                    db:
                      default: admin
                      description: DB is the database the user is stored in. Defaults
                        to "admin"
                      type: string
                    name:
                      description: Name is the username of the user
                      type: string
                    passwordSecretRef:
                      description: PasswordSecretRef is a reference to the secret
                        containing this user's password
                      properties:
                        key:
                          description: Key is the key in the secret storing this password.
                            Defaults to "password"
                          type: string
                        name:
                          description: Name is the name of the secret storing this
                            user's password
                          type: string
                      required:
                      - name
                      type: object
                    roles:
                      description: Roles is an array of roles assigned to this user
                      items:
                        description: Role is the database role this user should have
                        properties:
                          db:
                            description: DB is the database the role can act on
                            type: string
                          name:
                            description: Name is the name of the role
                            type: string
                        required:
                        - db
                        - name
                        type: object
                      type: array
                    scramCredentialsSecretName:
                      description: ScramCredentialsSecretName appended by string "scram-credentials"
                        is the name of the secret object created by the mongoDB operator
                        for storing SCRAM credentials These secrets names must be
                        different for each user in a deployment.
                      pattern: ^[a-z0-9]([-a-z0-9]*[a-z0-9])?(\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*$
                      type: string
                  required:
                  - name
                  - passwordSecretRef
                  - roles
                  - scramCredentialsSecretName
                  type: object
                type: array
              version:
                description: Version defines which version of MongoDB will be used
                type: string
            required:
            - security
            - type
            - users
            type: object
          status:
            description: MongoDBCommunityStatus defines the observed state of MongoDB
            properties:
              currentMongoDBArbiters:
                type: integer
              currentMongoDBMembers:
                type: integer
              currentStatefulSetArbitersReplicas:
                type: integer
              currentStatefulSetReplicas:
                type: integer
              message:
                type: string
              mongoUri:
                type: string
              phase:
                type: string
              version:
                type: string
            required:
            - currentMongoDBMembers
            - currentStatefulSetReplicas
            - mongoUri
            - phase
            type: object
        type: object
    served: true
    storage: true
    subresources:
      status: {}
status:
  acceptedNames:
    kind: ""
    plural: ""
  conditions: []
  storedVersions: []
```

 ```
vincenzocalia@vincenzos-MacBook-Air server-node % kubectl get crd/mongodbcommunity.mongodbcommunity.mongodb.com
NAME                                            CREATED AT
mongodbcommunity.mongodbcommunity.mongodb.com   2022-07-28T13:09:48Z
vincenzocalia@vincenzos-MacBook-Air server-node % kubectl get role mongodb-kubernetes-operator            
NAME                          CREATED AT
mongodb-kubernetes-operator   2022-07-28T13:10:28Z
vincenzocalia@vincenzos-MacBook-Air server-node % kubectl get role mongodb-kubernetes-operator --namespace default

NAME                          CREATED AT
mongodb-kubernetes-operator   2022-07-28T13:10:28Z
vincenzocalia@vincenzos-MacBook-Air server-node % kubectl get rolebinding mongodb-kubernetes-operator             
NAME                          ROLE                               AGE
mongodb-kubernetes-operator   Role/mongodb-kubernetes-operator   4h34m
vincenzocalia@vincenzos-MacBook-Air server-node % kubectl get rolebinding mongodb-kubernetes-operator --namespace default
NAME                          ROLE                               AGE
mongodb-kubernetes-operator   Role/mongodb-kubernetes-operator   4h34m
vincenzocalia@vincenzos-MacBook-Air server-node % kubectl get serviceaccount mongodb-kubernetes-operator         
NAME                          SECRETS   AGE
mongodb-kubernetes-operator   0         4h34m
vincenzocalia@vincenzos-MacBook-Air server-node % kubectl get serviceaccount mongodb-kubernetes-operator --namespace default
NAME                          SECRETS   AGE
mongodb-kubernetes-operator   0         4h34m
vincenzocalia@vincenzos-MacBook-Air server-node % kubectl get pods            
NAME                                           READY   STATUS    RESTARTS   AGE
mongodb-kubernetes-operator-7646658db4-8cnvn   1/1     Running   0          63m
vincenzocalia@vincenzos-MacBook-Air server-node % kubectl get pods --namespace default
NAME                                           READY   STATUS    RESTARTS   AGE
mongodb-kubernetes-operator-7646658db4-8cnvn   1/1     Running   0          63m
vincenzocalia@vincenzos-MacBook-Air server-node % kubectl get sts
No resources found in default namespace.
vincenzocalia@vincenzos-MacBook-Air server-node % kubectl api-resources
NAME                              SHORTNAMES   APIVERSION                             NAMESPACED   KIND
bindings                                       v1                                     true         Binding
componentstatuses                 cs           v1                                     false        ComponentStatus
configmaps                        cm           v1                                     true         ConfigMap
endpoints                         ep           v1                                     true         Endpoints
events                            ev           v1                                     true         Event
limitranges                       limits       v1                                     true         LimitRange
namespaces                        ns           v1                                     false        Namespace
nodes                             no           v1                                     false        Node
persistentvolumeclaims            pvc          v1                                     true         PersistentVolumeClaim
persistentvolumes                 pv           v1                                     false        PersistentVolume
pods                              po           v1                                     true         Pod
podtemplates                                   v1                                     true         PodTemplate
replicationcontrollers            rc           v1                                     true         ReplicationController
resourcequotas                    quota        v1                                     true         ResourceQuota
secrets                                        v1                                     true         Secret
serviceaccounts                   sa           v1                                     true         ServiceAccount
services                          svc          v1                                     true         Service
mutatingwebhookconfigurations                  admissionregistration.k8s.io/v1        false        MutatingWebhookConfiguration
validatingwebhookconfigurations                admissionregistration.k8s.io/v1        false        ValidatingWebhookConfiguration
customresourcedefinitions         crd,crds     apiextensions.k8s.io/v1                false        CustomResourceDefinition
apiservices                                    apiregistration.k8s.io/v1              false        APIService
controllerrevisions                            apps/v1                                true         ControllerRevision
daemonsets                        ds           apps/v1                                true         DaemonSet
deployments                       deploy       apps/v1                                true         Deployment
replicasets                       rs           apps/v1                                true         ReplicaSet
statefulsets                      sts          apps/v1                                true         StatefulSet
tokenreviews                                   authentication.k8s.io/v1               false        TokenReview
localsubjectaccessreviews                      authorization.k8s.io/v1                true         LocalSubjectAccessReview
selfsubjectaccessreviews                       authorization.k8s.io/v1                false        SelfSubjectAccessReview
selfsubjectrulesreviews                        authorization.k8s.io/v1                false        SelfSubjectRulesReview
subjectaccessreviews                           authorization.k8s.io/v1                false        SubjectAccessReview
horizontalpodautoscalers          hpa          autoscaling/v2                         true         HorizontalPodAutoscaler
cronjobs                          cj           batch/v1                               true         CronJob
jobs                                           batch/v1                               true         Job
certificatesigningrequests        csr          certificates.k8s.io/v1                 false        CertificateSigningRequest
leases                                         coordination.k8s.io/v1                 true         Lease
endpointslices                                 discovery.k8s.io/v1                    true         EndpointSlice
events                            ev           events.k8s.io/v1                       true         Event
flowschemas                                    flowcontrol.apiserver.k8s.io/v1beta2   false        FlowSchema
prioritylevelconfigurations                    flowcontrol.apiserver.k8s.io/v1beta2   false        PriorityLevelConfiguration
mongodbcommunity                  mdbc         mongodbcommunity.mongodb.com/v1        true         MongoDBCommunity
ingressclasses                                 networking.k8s.io/v1                   false        IngressClass
ingresses                         ing          networking.k8s.io/v1                   true         Ingress
networkpolicies                   netpol       networking.k8s.io/v1                   true         NetworkPolicy
runtimeclasses                                 node.k8s.io/v1                         false        RuntimeClass
poddisruptionbudgets              pdb          policy/v1                              true         PodDisruptionBudget
podsecuritypolicies               psp          policy/v1beta1                         false        PodSecurityPolicy
clusterrolebindings                            rbac.authorization.k8s.io/v1           false        ClusterRoleBinding
clusterroles                                   rbac.authorization.k8s.io/v1           false        ClusterRole
rolebindings                                   rbac.authorization.k8s.io/v1           true         RoleBinding
roles                                          rbac.authorization.k8s.io/v1           true         Role
priorityclasses                   pc           scheduling.k8s.io/v1                   false        PriorityClass
csidrivers                                     storage.k8s.io/v1                      false        CSIDriver
csinodes                                       storage.k8s.io/v1                      false        CSINode
csistoragecapacities                           storage.k8s.io/v1                      true         CSIStorageCapacity
storageclasses                    sc           storage.k8s.io/v1                      false        StorageClass
volumeattachments                              storage.k8s.io/v1                      false        VolumeAttachment
```
# manager.yaml
```
apiVersion: apps/v1
kind: Deployment
metadata:
  annotations:
    email: support@mongodb.com
  labels:
    owner: mongodb
  name: mongodb-kubernetes-operator
spec:
  replicas: 1
  selector:
    matchLabels:
      name: mongodb-kubernetes-operator
  strategy:
    rollingUpdate:
      maxUnavailable: 1
    type: RollingUpdate
  template:
    metadata:
      labels:
        name: mongodb-kubernetes-operator
    spec:
      affinity:
        podAntiAffinity:
          requiredDuringSchedulingIgnoredDuringExecution:
          - labelSelector:
              matchExpressions:
              - key: name
                operator: In
                values:
                - mongodb-kubernetes-operator
            topologyKey: kubernetes.io/hostname
      containers:
      - command:
        - /usr/local/bin/entrypoint
        env:
        - name: WATCH_NAMESPACE
          value: "*"
          # valueFrom:
          #   fieldRef:
          #     fieldPath: metadata.namespace
        - name: POD_NAME
          valueFrom:
            fieldRef:
              fieldPath: metadata.name
        - name: OPERATOR_NAME
          value: mongodb-kubernetes-operator
        - name: AGENT_IMAGE
          value: quay.io/mongodb/mongodb-agent:11.12.0.7388-1
        - name: VERSION_UPGRADE_HOOK_IMAGE
          value: quay.io/mongodb/mongodb-kubernetes-operator-version-upgrade-post-start-hook:1.0.4
        - name: READINESS_PROBE_IMAGE
          value: quay.io/mongodb/mongodb-kubernetes-readinessprobe:1.0.9
        - name: MONGODB_IMAGE
          value: mongo
        - name: MONGODB_REPO_URL
          value: docker.io
        image: quay.io/mongodb/mongodb-kubernetes-operator:0.7.4
        imagePullPolicy: Always
        name: mongodb-kubernetes-operator
        resources:
          limits:
            cpu: 1100m
            memory: 1Gi
          requests:
            cpu: 500m
            memory: 200Mi
        securityContext:
          readOnlyRootFilesystem: true
          runAsUser: 2000
          allowPrivilegeEscalation: false
      securityContext:
        seccompProfile:
          type: RuntimeDefault
      serviceAccountName: mongodb-kubernetes-operator

```
# cluster_role_binding.yaml
```
kind: ClusterRoleBinding
apiVersion: rbac.authorization.k8s.io/v1
metadata:
  name: mongodb-kubernetes-operator
subjects:
- kind: ServiceAccount
  namespace: default
  name: mongodb-kubernetes-operator
roleRef:
  kind: ClusterRole
  name: mongodb-kubernetes-operator
  apiGroup: rbac.authorization.k8s.io
````

**Comment by vinnytwice** [comment]
So I could easily test settings and remove applied files
added all yaml files in the skaffold deploy: kubectl: manifests:
  
```
deploy:
  kubectl:
    manifests:
      # # - ./infra/k8s/*
      
      - /Users/vincenzocalia/mongodb-kubernetes-operator/config/crd/bases/mongodbcommunity.mongodb.com_mongodbcommunity.yaml
      - /Users/vincenzocalia/mongodb-kubernetes-operator/config/rbac/role_binding_database.yaml
      - /Users/vincenzocalia/mongodb-kubernetes-operator/config/rbac/role_binding.yaml
      - /Users/vincenzocalia/mongodb-kubernetes-operator/config/rbac/role_database.yaml
      - /Users/vincenzocalia/mongodb-kubernetes-operator/config/rbac/role.yaml
      - /Users/vincenzocalia/mongodb-kubernetes-operator/config/rbac/service_account_database.yaml
      - /Users/vincenzocalia/mongodb-kubernetes-operator/config/rbac/service_account.yaml
      - /Users/vincenzocalia/mongodb-kubernetes-operator/config/manager/manager.yaml
```




As first test I just added a `namespace: default` in `/config/rbac/role_binding_database.yaml`, `/config/rbac/role_binding.yaml`,
`/config/rbac/role_database.yaml`,
`/config/rbac/role.yaml`,
`/config/rbac/service_account_database.yaml`,
`/config/rbac/service_account.yaml` files.

No difference..

I found that I wrongly edited `config/crd/bases/mongodbcommunity.mongodb.com_mongodbcommunity.yaml` file  so I put back the controller-gen.kubebuilder.io/version: to correct `v0.4.1` value, dough it didn't solve my issue.  Desired Pod are still not been scheduled.
```
metadata:
  annotations:
    controller-gen.kubebuilder.io/version: v0.4.1 #v0.7.4  
  creationTimestamp: null
  name: mongodbcommunity.mongodbcommunity.mongodb.com
```
I decided to try and edited back also the manager file /config/manager/manager.yaml to leave operator watching the the specified namespace instead of any as I set it initially
```
env:
        - name: WATCH_NAMESPACE
          # value: "*"
          valueFrom:
            fieldRef:
              fieldPath: metadata.namespace
``` 
and BINGO! now pod get scheduled..not sure why they didn't with `value: "*"` but the do start now.

Still I get the `error: This apiVersion and/or kind does not reference a schema known by Cloud Code. Please ensure you are using a valid apiVersion and kind.` warning dough..

**Comment by dan-mckean** [comment]
Hi, I'm Dan and I'm the Product Manager for MongoDB's support of Kubernetes.

I'm doing some work right now to try and identify how the Community Operator is being used. The Community Operator is something I inherited when I started at MongoDB, but it doesn't get as much attention from us as we'd like and we're trying to understand how it's used in order to establish its future. It will help us prioritize future issues and PRs raised by the community 🙂

Here's a super short survey (it's much easier for us to review all the feedback that way!): [https://docs.google.com/forms/d/e/1FAIpQLSfwrwyxBSlUyJ6AmC-eYlgW_3JEdfA48SB2i5--_WpiynMW2w/viewform?usp=sf_link](https://docs.google.com/forms/d/e/1FAIpQLSfwrwyxBSlUyJ6AmC-eYlgW_3JEdfA48SB2i5--_WpiynMW2w/viewform?usp=sf_link)

 If you'd rather email me instead: [dan.mckean@mongodb.com](mailto:dan.mckean@mongodb.com?subject=MongoDB%20Community%20Operator%20feedback)

Thank you in advance!
Dan
